ovs/lib/albapreset.py

```diff
@@ -0,0 +1,201 @@
+# Copyright (C) 2016 iNuron NV
+#
+# This file is part of Open vStorage Open Source Edition (OSE),
+# as available from
+#
+#      http://www.openvstorage.org and
+#      http://www.openvstorage.com.
+#
+# This file is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License v3 (GNU AGPLv3)
+# as published by the Free Software Foundation, in version 3 as it comes
+# in the LICENSE.txt file of the Open vStorage OSE distribution.
+#
+# Open vStorage is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY of any kind.
+
+"""
+AlbaPresetController module
+"""
+
+import os
+import re
+import json
+import random
+import tempfile
+from ovs.celery_run import celery
+from ovs.dal.hybrids.albabackend import AlbaBackend
+from ovs.extensions.db.arakoon.ArakoonInstaller import ArakoonInstaller
+from ovs.extensions.generic.configuration import Configuration
+from ovs.extensions.plugins.albacli import AlbaCLI
+from ovs.lib.alba import AlbaController
+from ovs.lib.helpers.toolbox import Toolbox
+from ovs.log.log_handler import LogHandler
+
+
+class AlbaPresetController(object):
+    """
+    Contains all BLL related to ALBA presets
+    """
+    _logger = LogHandler.get('lib', name='albapreset')
+
+    @staticmethod
+    @celery.task(name='alba.add_preset')
+    def add_preset(alba_backend_guid, name, compression, policies, encryption, fragment_size=None):
+        """
+        Adds a preset to Alba
+        :param alba_backend_guid: Guid of the ALBA backend
+        :type alba_backend_guid: str
+        :param name: Name of the preset
+        :type name: str
+        :param compression: Compression type for the preset (none | snappy | bz2)
+        :type compression: str
+        :param policies: Policies for the preset
+        :type policies: list
+        :param encryption: Encryption for the preset (none | aes-cbc-256 | aes-ctr-256)
+        :type encryption: str
+        :param fragment_size: Size of a fragment in bytes (e.g. 1048576)
+        :type fragment_size: int
+        :return: None
+        """
+        # VALIDATIONS
+        if not re.match(Toolbox.regex_preset, name):
+            raise ValueError('Invalid preset name specified')
+
+        compression_options = ['snappy', 'bz2', 'none']
+        if compression not in compression_options:
+            raise ValueError('Invalid compression format specified, please choose from: "{0}"'.format('", "'.join(compression_options)))
+
+        encryption_options = ['aes-cbc-256', 'aes-ctr-256', 'none']
+        if encryption not in encryption_options:
+            raise ValueError('Invalid encryption format specified, please choose from: "{0}"'.format('", "'.join(encryption_options)))
+
+        if fragment_size is not None and (not isinstance(fragment_size, int) or not 0 < fragment_size <= 1024 ** 3):
```

Comment:
The last time I polled with Alba they said that the lowest fragment size is 16 Bytes.